setup

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# -e (-o errexit): exit immediately on error
+# -u (-o nounset): treat unset variables as erros
+# -o pipefail: propagate intermediate errors in pipelines
+set -euo pipefail
+
+main() {
+  print_step "Verifying that pip is available"
+  pip=$(command -v pip3 || command -v pip || echo -n "")
+  if [ -z "$pip" ]; then
+    print_error "Cannot find pip, aborting."
+    exit 1
+  fi
+
+  print_step "Installing and initializing pipenv"
+  $pip install pipenv
+
+  print_step "Installing Python dependencies"
+  pipenv install
+
+  print_step "Generating Ansible Vault secret"
+  pwgen -1 24|tee ~/.forem_selfhost_ansible_vault_password
+
+  print_step "Copying example inventory"
+  cp -v inventory/example/setup.yml inventory/forem/setup.yml
+
+  print_step "Generating Vault secrets"
+  echo -n $(pwgen -1 128) | ansible_vault vault_secret_key_base
+  echo -n $(xxd -g 2 -l 64 -p /dev/urandom | tr -d '\n') | ansible_vault vault_imgproxy_key
+  echo -n $(xxd -g 2 -l 64 -p /dev/urandom | tr -d '\n') | ansible_vault vault_imgproxy_salt
+  echo -n $(pwgen -1 128) | ansible_vault vault_forem_postgres_password
+  print_info "Please add these secrets at the appropiate places in inventory/forem/setup.yml"
+
+  print_info "Before continuing please activate your pipenv with the following command: pipenv shell"
+}
+
+ansible_vault() {
+  pipenv run ansible-vault encrypt_string --stdin-name $1
+}
+
+### OUTPUT HELPERS
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+YELLOW='\033[1;33m'
+NO_COLOR='\033[0m'
+
+color_print() {
+  local color=$1
+  shift
+  printf "\n${color}$*${NO_COLOR}\n"
```

Comment:
nitpick: adding ansi escape codes for color here makes assumptions about the stdout device that aren't _always_ true (rails logging is the same way, the log files are a mess if you open them as text without interpreting these escape codes, but the display in a terminal is almost always colored correctly).

Is it overly cautious to check (with `tput colors`?) that the output terminal supports these color codes before injecting them into the output? 

Please treat this as a light joke rather than racking your brain about it.

Reply:
@djuber only really cares about how this will look seven buffers deep in Emacs... colors tends to go plaid at that level.

Reply:
@djuber This was very much built for being run interactively in a terminal, so yes, there's a pretty strong assumption that ANSI escape sequences will be interpreted. But once I run out of other things to improve here I can definitely revisit the `tput colors` idea. 😃

Reply:
Ok, I went for the minimal approach, checking if file handle 1 is a terminal. If you're running a terminal that doesn't at least support 8 colors in 2021 you're probably used to seeing worse than the occasional ANSI escape sequence. Here's the before and after of piping into `less`:

```
# Before
ESC[0;32mVerifying that pip is availableESC[0m

ESC[0;32mInstalling and initializing pipenvESC[0m

# After
Verifying that pip is available

Installing and initializing pipenv
```